homeassistant/components/mqtt/__init__.py

```diff
@@ -245,7 +245,7 @@ async def async_check_config_schema(
                     message, _ = conf_util._format_config_error(
                         ex, domain, config, integration.documentation
                     )
-                    raise HomeAssistantError(message) from ex
+                    raise ServiceError(message) from ex
```

Comment:
Would make sense to subclass this as a ValidationError, that way we don't need that raise_on_failure

Reply:
The `raise_on_failure` is because the normal behavior is that it doesn't raise on failure but returns `None`. I want it to raise to prevent logging an error twice, what happens when we need to raise after all. So we could leave `raise_on_failure` away in this case, but than  the change would become breaking. May be that is worth it?

Reply:
Right. I think it should just always raise, with a specific enough exception, so it can be managed properly further out.

Reply:
My concern is that it will be breaking and custom integrations might break as well. So may be we should implement it under a different name and deprecate the old method with an issue registered?

Reply:
Sure. Would be an easy small wrapper. That catch the exception and return none instead.

Reply:
done